### PR TITLE
feat: add interactive persona creation with template selection

### PR DESCRIPTION
**Pull request by clusterfudge**
## Summary
Adds interactive persona creation prompts that allow users to optionally base new personas on built-in templates.

## User Experience
When a user starts silica with a non-existent persona, they are now prompted to create it:

```
$ silica --persona my-new-persona

The persona 'my-new-persona' doesn't exist yet.
Would you like to base it on a built-in persona template? [y/n]: y

Available persona templates:
  1. basic_agent - General purpose assistant with access to various tools
  2. deep_research_agent - Research and comprehensive document creation specialist
  3. autonomous_engineer - Autonomous software engineering and development agent
  4. blank - Start with no template

Choice [1-4]: 3

Creating persona 'my-new-persona' based on 'autonomous_engineer'
✓ Created persona directory: ~/.silica/personas/my-new-persona
✓ Created persona file: ~/.silica/personas/my-new-persona/persona.md
You can edit ~/.silica/personas/my-new-persona/persona.md to customize your persona.
```

### New Functions in `silica/developer/personas/__init__.py`
- `get_builtin_descriptions()` - Returns dict of persona names to descriptions
- `get_builtin_prompt(name)` - Returns prompt text for a built-in persona
- `create_persona_directory(name, base_prompt)` - Creates persona dir with persona.md
- `persona_exists(name)` - Checks if persona directory and file exist

### Updates to `silica/developer/hdev.py`
- Added `ensure_persona_exists()` function with interactive prompts
- Integrated persona creation check before agent starts
- Handles user cancellation gracefully (Ctrl+C)

### New Test File
- `tests/developer/test_persona_creation.py` with 13 comprehensive tests
- Tests all creation scenarios, cancellation, idempotency, etc.

## Benefits
✅ **User-friendly** - Clear prompts guide users through persona creation
✅ **Flexible** - Can use templates or start blank  
✅ **Safe** - Never overwrites existing persona.md files
✅ **Discoverable** - Shows available templates with descriptions
✅ **Cancelable** - Respects Ctrl+C at any point

## Testing
- ✅ All 13 new tests pass
- ✅ All 620 existing tests pass
- ✅ Pre-commit hooks pass (ruff, autoflake, ruff-format)
- ✅ Handles edge cases (cancellation, overwrites, invalid input)

## Documentation
The persona.md file serves as documentation for the persona's behavior and can be edited by users to customize their personas.

### CODE REVIEW

File: silica/developer/hdev.py [hunk 1]
```diff
@@ -32,12 +32,143 @@
...
 from prompt_toolkit.completion import Completer, WordCompleter, Completion
+from prompt_toolkit import prompt as pt_prompt
+from prompt_toolkit.validation import Validator, ValidationError
 
 
 SANDBOX_MODE_MAP = {mode.name.lower(): mode for mode in SandboxMode}
 SANDBOX_MODE_MAP["dwr"] = SandboxMode.ALLOW_ALL
 
 
+def ensure_persona_exists(persona_name: str, console: Console) -> bool:
```
[the remark this answers is not in the document]
Comment (clusterfudge) [reply silica/developer/hdev.py]: actually, change it to get_or_create persona, returns a Persona, and this is the method that is called by the entrypoint in hdev.

**Review by clusterfudge — COMMENTED** [review]
some feedback
